portal/src/main/webapp/js/src/oncoprint/setup.js

```diff
@@ -886,6 +888,44 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 	    return done.promise();
 	};
 	
+	var HEATMAP_RULE_SET_PARAMS = {
+		'type': 'gradient',
+		'legend_label': "Expression heatmap",
+		'value_key': 'profile_data',
+		'value_range': [-3, 3],
+		'colors': [[0,0,255,1],[0,0,0,1],[255,0,0,1]],
+		'value_stop_points': [-3,0,3],
+		'null_color': 'rgba(224,224,224,1)'
+	};
+	
+	/**
+	 * Makes a track reuse the legend of any previous heatmap tracks.
+	 *
+	 * If any tracks previously given to this function still
+	 * exist, make the new track share the rule set and legend
+	 * of the existing ones.
+	 *
+	 * @param {number} track_id - ID of a track to use the shared legend
+	 */
+	var useExistingHeatmapRuleSet = (function() {
```

Comment:
`getAnyHeatmapTrackId` would not have to return *any* heatmap track ID, but specifically one which shares the same legend. That is, an RNA or protein expression heatmap track either added via the Heatmap menu (tracked in the list it used before) or one added by expanding gene set tracks (not tracked in the same list, as they share no other functionality with the former), but not a gene set heatmap track (different legend). Because of all this, I thought it would be confusing not to make the function explicitly about sharing the ruleset/legend between them.